activities.py

```diff
@@ -1,150 +1,136 @@
-        except yaml.YAMLError as e:
-            raise ValueError("Failed to load YAML: Invalid YAML format") from e
-
-    def log_error(self, message: str, key: str = "", line: int = None, fixable=False):
-        tofix = " (Use --fix to fix.)" if fixable else ""
-        if line is not None:
-            self.errors.append(f"Error at line {line} in key '{key}': {message}{tofix}")
-        else:
-            self.errors.append(f"Error: {message}{tofix}")
-
-    def validate_top_level_keys(self, data: Dict[str, Any], line_map: Dict[int, str]):
-        keys = list(data.keys())
-        modified = False
+                data = self.yaml.load(file)
+                return data
+        except Exception as e:
+            raise ValueError(f"Failed to load YAML: {e}") from e
 
-        # Check for non-string keys
-        non_string_keys = [key for key in keys if not isinstance(key, str)]
-        if non_string_keys:
-            for key in non_string_keys:
-                line_num = next((line for line, content in line_map.items() if str(key) in content), None)
-                self.log_error("Top-level keys must be strings", key, line_num)
+    def log_error(self, message: str, key: str = ""):
+        self.errors.append(f"Error in key '{key}': {message}")
 
-        # Check alphabetical order
+    def validate_top_level_keys(self, data: CommentedMap):
+        keys = list(data.keys())
         if keys != sorted(keys):
             if self.fix:
-                data = dict(sorted(data.items()))
-                modified = True
+                self.modified_data = CommentedMap(sorted(data.items()))
             else:
-                self.log_error("Top-level keys must be in alphabetical order.", fixable=True)
-
-        # Check for unique keys
-        unique_keys = set(keys)
-        if len(keys) != len(unique_keys):
-            duplicates = [key for key in keys if keys.count(key) > 1]
-            for key in set(duplicates):
-                line_num = next((line for line, content in line_map.items() if key in content), None)
-                self.log_error("Top-level keys must be unique", key, line_num)
-
-        if self.fix and modified:
-            self.modified_data = data  # Track modified data for saving later
-
-    def validate_item(self, item: Dict[str, Any], key: str, line: int):
-        modified = False
-
-        # Validate issue (required and unique)
+                self.log_error("Top-level keys must be in alphabetical order.")
+
+    def validate_literal_block(self, data: CommentedMap, key: str, field: str):
+        value = data.get(key, {}).get(field)
+        if value is not None:
+            if not isinstance(value, LiteralScalarString):
+                if self.fix:
+                    data[key][field] = LiteralScalarString(value)
+                else:
+                    self.log_error(f"'{field}' must use literal block syntax (|).", key)
+            elif not value.endswith('\n'):
```

Comment:
Is it even possible to have a value that doesn't have a trailing `\n`?

Reply:
Yes:
```
foo: |-
  bar
```